js/windowMgr.js

```diff
@@ -174,6 +175,28 @@ function doCreateMainWindow(initialUrl, initialBounds) {
 
     mainWindow.on('closed', destroyAllWindows);
 
+    // Manage File Downloads
+    mainWindow.webContents.session.on('will-download', (event, item, webContents) => {
+        // When download is in progress, send necessary data to indicate the same
+        mainWindow.send('downloadProgress');
+
+        // Send file path when download is complete
+        item.once('done', (event, state) => {
+            if (state === 'completed') {
+                let data = {
+                    _id: getGuid(),
+                    filePath: item.getSavePath(),
+                    total: filesize(item.getTotalBytes()),
+                    fileName: item.getFilename(),
+                    mimeType: item.getMimeType()
+                };
+                mainWindow.send('download-completed', data);
+            } else {
+                console.log(`Download failed: ${state}`);
```

Comment:
if download fails, don't we have to send something to renderer to show some failed state?  will it keep downloading forever if we don't?